File: codeforces/solved/144A.cc
#include <iostream>

using namespace std;

int main() {
  int n;
  cin >> n;
  int min = 200;
  int max = -1;
  int minpos, maxpos;
  int e;
  for(int i=0; i<n; ++i) {
    cin >> e;
    if(e <= min) {
      min = e;
      minpos = i;
    }

    if(e > max) {
      max = e;
      maxpos = i;
    }
  }

  int ans = maxpos + (n - minpos) - 1;

  if(minpos < maxpos)
    ans -= 1;

  // cout << "-------------------------------" << endl;
  // cout << "min=" << min << endl;
  // cout << "max=" << max << endl;
  // cout << "minpos=" << minpos << endl;
  // cout << "maxpos=" << maxpos << endl;
  // cout << "swap=" << (minpos < maxpos) << endl;
  cout << ans;

  return 0;
}

File: codeforces/solved/144A.rb
def order(n,a)
  minmax = a.minmax
  min = minmax.shift
  max = minmax.shift

  #As the smaller soldier has to be at the end of the line we need to look for the closest one to that pos
  min_pos = a.rindex(min)
  # The highest soldier has to be first that's why we start looking from the beginig of the line
  max_pos = a.index(max)

  tot_sec = 0
  #Checking if the soldiers have to be swapped between them
  if min_pos < max_pos
    # Has to be -2 because of the array index and one less to the end of line,
    # that's because the soldiers are swapped between them and we consider that time on the max_pos
    tot_sec = max_pos + ((n-2) - min_pos)
  else
    tot_sec = max_pos + ((n-1) - min_pos)
  end

  tot_sec
end

n = gets.to_i
a = gets.split(/\s/).map(&:to_i)
puts order(n,a)